### PR TITLE
feat: use Agent v2 API for Service Banner

### PR DESCRIPTION
**Pull request by spikecurtis**
Agent uses the v2 API for the service banner, rather than the v1 HTTP API.

One of several for #10534

### CODE REVIEW

**Comment by spikecurtis** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/11806?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `main`
  * **#11792** <a href="https://app.graphite.dev/github/pr/coder/coder/11792?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#11736** <a href="https://app.graphite.dev/github/pr/coder/coder/11736?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **#11768** <a href="https://app.graphite.dev/github/pr/coder/coder/11768?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **#11769** <a href="https://app.graphite.dev/github/pr/coder/coder/11769?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **#11770** <a href="https://app.graphite.dev/github/pr/coder/coder/11770?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **#11806** <a href="https://app.graphite.dev/github/pr/coder/coder/11806?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @spikecurtis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: agent/agenttest/client.go [hunk 1]
```diff
@@ -254,3 +246,73 @@ type closeFunc func() error
 func (c closeFunc) Close() error {
 	return c()
 }
+
+type FakeAgentAPI struct {
```
Comment (deansheather) [inline agent/agenttest/client.go:250]: Could you generate this?

**Review by deansheather — APPROVED** [review]

File: agent/agenttest/client.go (hunk 1 above)
Comment (spikecurtis) [reply agent/agenttest/client.go:250]: I could generate it as a Mock, but that won't fit as easily into our existing tests.

**Comment by spikecurtis** [comment]
### Merge activity

* **Jan 29, 10:44 PM**: @spikecurtis merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/11806).

**Pull request merged**